shared/docs/react-native/profiling.md

```diff
@@ -0,0 +1,280 @@
+  * It has a lot of good stuff and ideas. The rest of this will assume you've read
+the prereqs
+
+* [Parashuram's Dive into RN performance](http://blog.nparashuram.com/2018/11/react-native-performance-playbook-part-i.html)
+  * This is mostly android specific, but still generally useful
+  * Shows by example some RN profiling tools that aren't well documented
+
+* [React's own profiler](https://reactjs.org/blog/2018/09/10/introducing-the-react-profiler.html)
+  * This is react dev tool's own profiler. It includes more react specific
+    information than just user timings. If you're optimizing a component this
+    is what you want.
+
+* [Chrome timeline](https://developers.google.com/web/tools/chrome-devtools/evaluate-performance/reference)
+  * Look at this for a basic understanding of how to use the chrome timeline
+    view. We use this to look at user timings. We can visualize the perfomance
+    calls with this.
+
+* [JS Performance API](https://developer.mozilla.org/en-US/docs/Web/API/Performance)
+  * Basically understand how to use `performance.mark` and
+    `performance.measure` this is useful for user timing stuff.
+
+## Considerations when remote debugging
+When using the remote debugging, remember that your computer is much faster at
+running JS than your phone. On android this is pretty noticeable. There is a
+setting to throttle your CPU when you profile the app in the timeline view, and
+I find that useful in trying to get closer to what the android phone would do.
+
+## User Timings
+`Performance.mark` and `Performance.measure` are useful tools to see how long
+things take. Chrome's user timings will also display these in the timeline
+view. Thanks to Nojima, we already have quite a bit. Turn on userTimings in
+local-debug and you'll get user timings for engine calls, saga calls, and
+reducer calls. Sometimes the saga calls can be unwieldy so it may help to
+manually create a filter on which types of saga calls will get
+profiled. You can modify `util/user-timings` to do this.
+
+Make sure you have a performance polyfill setup or it will fail when running on
+the native device (at least on android)
+
+## Bundling strategies
+[React Native's Performace
+doc](https://facebook.github.io/react-native/docs/performance) has a section on
+this. As of my latest test, there isn't a real difference in using this, but
+this will likely change when we stop loading so many files on startup.
+
+There is an issue with indexed-ram-bundling on [android](https://github.com/facebook/react-native/issues/21282)
+
+## ReactNative.Systrace
+
+Api [here](https://facebook.github.io/react-native/docs/systrace)
+
+This is super useful in analyzing startup time performance cost. You can use
+this to see how long requires are taking. But there's a trick. You have to
+change the implementation of a couple methods for it to work properly.
+
+Add this to your entry file to enable Systrace (you can also enable it via RN
```

Comment:
I can't get this to work, `Can't find variable: performance` (I put it in `index.android`). More specifics for where this should be? Also, button path for enabling via RN devtools would be nice.

Reply:
What do you mean button path?

Reply:
I forgot to mention how to add the performance polyfill! which is needed if you aren't connected to a remote debugger.

Reply:
Ah, that'd explain it. By button path I mean I don't know how to enable it via RN devtools.

Reply:
Ah yeah, I never used it, but it's there: 

![](https://facebook.github.io/react-native/docs/assets/DeveloperMenu.png)

I'll remove that sentence

Reply:
Ah that option isn't on android